internal/provider/buildingblock.go

```diff
@@ -1,43 +1,42 @@
 package provider
 
-import "time"
-
 type MeshBuildingBlock struct {
-	ApiVersion string                    `json:"apiVersion"`
-	Kind       string                    `json:"kind"`
-	Metadata   MeshBuildingBlockMetadata `json:"metadata"`
-	Spec       MeshBuildingBlockSpec     `json:"spec"`
-	Status     MeshBuildingBlockStatus   `json:"status"`
+	ApiVersion string                    `json:"apiVersion" tfsdk:"api_version"`
+	Kind       string                    `json:"kind" tfsdk:"kind"`
+	Metadata   MeshBuildingBlockMetadata `json:"metadata" tfsdk:"metadata"`
+	Spec       MeshBuildingBlockSpec     `json:"spec" tfsdk:"spec"`
+	Status     MeshBuildingBlockStatus   `json:"status" tfsdk:"status"`
 }
 
 type MeshBuildingBlockMetadata struct {
-	Uuid                string    `json:"uuid"`
-	DefinitionUuid      string    `json:"definitionUuid"`
-	DefinitionVersion   int64     `json:"definitionVersion"`
-	TenantIdentifier    string    `json:"tenantIdentifier"`
-	CreatedOn           time.Time `json:"createdOn"`
-	MarkedForDeletionOn time.Time `json:"markedForDeletionOn"`
-	MarkedForDeletionBy string    `json:"markedForDeletionBy"`
+	Uuid                string  `json:"uuid" tfsdk:"uuid"`
+	DefinitionUuid      string  `json:"definitionUuid" tfsdk:"definition_uuid"`
+	DefinitionVersion   int64   `json:"definitionVersion" tfsdk:"definition_version"`
+	TenantIdentifier    string  `json:"tenantIdentifier" tfsdk:"tenant_identifier"`
+	ForcePurge          bool    `json:"forcePurge" tfsdk:"force_purge"`
+	CreatedOn           string  `json:"createdOn" tfsdk:"created_on"`
+	MarkedForDeletionOn *string `json:"markedForDeletionOn" tfsdk:"marked_for_deletion_on"`
+	MarkedForDeletionBy *string `json:"markedForDeletionBy" tfsdk:"marked_for_deletion_by"`
 }
 
 type MeshBuildingBlockSpec struct {
-	DisplayName          string                `json:"displayName"`
-	Inputs               []MeshBuildingBlockIO `json:"inputs"`
-	ParentBuildingBlocks []ParentBuildingBlock `json:"parentBuildingBlocks"`
+	DisplayName          string                    `json:"displayName" tfsdk:"display_name"`
+	Inputs               []MeshBuildingBlockIO     `json:"inputs" tfsdk:"inputs"`
+	ParentBuildingBlocks []MeshBuildingBlockParent `json:"parentBuildingBlocks" tfsdk:"parent_building_blocks"`
 }
 
 type MeshBuildingBlockIO struct {
-	Key       string `json:"key"`
-	Value     any    `json:"value"`
-	ValueType string `json:"valueType"`
+	Key       string `json:"key" tfsdk:"key"`
+	Value     string `json:"value" tfsdk:"value"`
```

Comment:
we support int and bool as well in the backend.

Reply:
Hm. according to the docs we also have FILE, LIST, SINGLE_SELECT.